tests/Functional/AutoincrementColumnTest.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Doctrine\DBAL\Tests\Functional;
+
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+
+use function strtolower;
+
+class AutoincrementColumnTest extends FunctionalTestCase
+{
+    /** @var bool */
+    private static $tableCreated = false;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        if (! self::$tableCreated) {
+            $table = new Table('auto_increment_table');
+            $table->addColumn('id', 'integer', ['autoincrement' => true]);
+            $table->addColumn('test_int', 'integer');
+            $table->setPrimaryKey(['id']);
+
+            $sm = $this->connection->getSchemaManager();
+            $sm->createTable($table);
```

Comment:
Please use `dropAndCreateTable()` and create the table unconditionally. This is effectively what truncation does.

Reply:
OK, thanks.